// guxta-api/index.js
// const de download 
const { 
aiovideodl,
joox, 
soundcloud, 
tiktok, 
umma, 
ringtone, 
twitter, 
youtube, 
igdl, 
igstory, 
dafontDown, 
anoboydl
} = require('./lib/social-media/download.js')
// const de animes
const {
pinterestVideoV2
} = require('./lib/social-media/pinvid.js')
const {
hentaistube, 
pornhub, 
assistithtdl, 
assistitht, 
pornogratis
} = require('./lib/search/api.js')
// const de animes
const { 
konachan, 
anime, 
manga, 
characters, 
myanimelist, 
quotesanime, 
anoboy 
} = require('./lib/anime/anime.js')
// const de pesquisas
const { 
lirik, 
linkwa, 
dafontSearch, 
GSMArena, 
emoji 
} = require('./lib/search/search.js')
// const de notícias 
const { 
kompasnews, 
inews, 
merdekanews 
} = require('./lib/news/news.js')
// const comic
const { 
wattpad,
wattpaduser, 
komiku, komikuv2 
} = require('./lib/comic/comic.js')
//const de stalkers
const { 
igstalk 
} = require('./lib/stalker/stalker.js')

//===================================
// DOWNLOADS :)
module.exports.aiovideodl = aiovideodl
module.exports.joox = joox
module.exports.soundcloud = soundcloud
module.exports.tiktok = tiktok
module.exports.umma = umma
module.exports.ringtone = ringtone
module.exports.twitter = twitter
module.exports.youtube = youtube
module.exports.igdl = igdl
module.exports.igstory = igstory
module.exports.dafontDown = dafontDown
module.exports.anoboydl = anoboydl
module.exports.pinterestVideoV2 = pinterestVideoV2

//===================================
// ANIMES :)
module.exports.konachan = konachan
module.exports.anime = anime
module.exports.manga = manga
module.exports.characters = characters
module.exports.myanimelist = myanimelist
module.exports.quotesanime = quotesanime
module.exports.anoboy = anoboy

//===================================
// PESQUISAS :)
module.exports.lirik = lirik
module.exports.linkwa = linkwa
module.exports.dafontSearch = dafontSearch
module.exports.GSMArena = GSMArena
module.exports.emoji = emoji
module.exports.hentaistube = hentaistube
module.exports.pornhub = pornhub
module.exports.assistithtdl = assistithtdl
module.exports.assistitht = assistitht
module.exports.pornogratis = pornogratis

//===================================
// NOTÍCIAS :)
module.exports.kompasnews = kompasnews
module.exports.inews = inews
module.exports.merdekanews = merdekanews

//===================================
// COMIC :)
module.exports.wattpad = wattpad
module.exports.wattpaduser = wattpaduser
module.exports.komiku = komiku
module.exports.komikuv2 = komikuv2

//===================================
// STALKER
module.exports.igstalk = igstalk
